cpp/lib/ome/xml/model/OriginalMetadataAnnotation.cpp

```diff
@@ -0,0 +1,237 @@
+
+#include <boost/format.hpp>
+
+#include <boost/format.hpp>
+
+#include <ome/xerces/Platform.h>
+
+#include <ome/xml/model/ModelException.h>
+#include <ome/xml/model/OMEModel.h>
+#include <ome/xml/model/OriginalMetadataAnnotation.h>
+
+using boost::format;
+
+namespace ome
+{
+  namespace xml
+  {
+    namespace model
+    {
+
+      namespace
+      {
+        const std::string MAP_NAMESPACE("http://www.openmicroscopy.org/Schemas/SA/" OME_MODEL_VERSION);
+        const std::string PAIRS_NAMESPACE("http://www.openmicroscopy.org/Schemas/OME/" OME_MODEL_VERSION);
+      }
+
+      OriginalMetadataAnnotation::OriginalMetadataAnnotation ():
+        ::ome::xml::model::OMEModelObject(),
+        XMLAnnotation(),
+        metadata()
+      {
+      }
+
+      OriginalMetadataAnnotation::OriginalMetadataAnnotation (const OriginalMetadataAnnotation& copy):
+        ::ome::xml::model::OMEModelObject(),
+        XMLAnnotation(copy),
+        metadata(copy.metadata)
+      {
+      }
+
+      OriginalMetadataAnnotation::OriginalMetadataAnnotation (const metadata_type& metadata):
+        ::ome::xml::model::OMEModelObject(),
+        XMLAnnotation(),
+        metadata(metadata)
+      {
+      }
+
+      OriginalMetadataAnnotation::OriginalMetadataAnnotation (xerces::dom::Element&        element,
+                          ::ome::xml::model::OMEModel& model):
+        XMLAnnotation(),
+        metadata()
+      {
+        update(element, model);
+      }
+
+      OriginalMetadataAnnotation::~OriginalMetadataAnnotation ()
+      {
+      }
+
+      const std::string&
+      OriginalMetadataAnnotation::elementName() const
+      {
+        static const std::string type("OriginalMetadataAnnotation");
+        return type;
+      }
+
+      bool
+      OriginalMetadataAnnotation::validElementName(const std::string& name) const
+      {
+        static const std::string expectedTagName("OriginalMetadataAnnotation");
+
+        return expectedTagName == name || XMLAnnotation::validElementName(name);
+      }
+
+      void
+      OriginalMetadataAnnotation::update(const xerces::dom::Element&  element,
+                                         ::ome::xml::model::OMEModel& model)
+      {
+        // Delegate checking of element type.
+        XMLAnnotation::update(element, model);
+
+        std::vector<xerces::dom::Element> XMLValue_nodeList = getChildrenByTagName(element, "Value");
+        if (XMLValue_nodeList.size() > 1)
+          {
+            format fmt("Value node list size %1% != 1");
+            fmt % XMLValue_nodeList.size();
+            throw ModelException(fmt.str());
+          }
+        else if (XMLValue_nodeList.size() != 0)
+          {
+            std::vector<xerces::dom::Element> OriginalMetadataValue_nodeList = getChildrenByTagName(XMLValue_nodeList.at(0), "OriginalMetadata");
+            if (OriginalMetadataValue_nodeList.size() > 1)
+              {
+                format fmt("Value node list size %1% != 1");
```

Comment:
For the original metadata exception do we want a different exception message string from the above so they are more easily distinguished?